src/daft-core/src/array/ops/comparison.rs

```diff
@@ -800,54 +799,86 @@ impl DaftLogical<&BooleanArray> for BooleanArray {
 }
 
 
-// Implementaing Daft Logic for Numeric values
-macro_rules! impl_daft_logical_for_integers {
-    ($type:ty) => {
-        impl DaftLogical<$type> for $type {
-            type Output = DaftResult<$type>;
-
-            fn and(&self, rhs: Self) -> Self::Output {
-                let result_bitmap = self
-                    .values()
-                    .iter()
-                    .zip(rhs.values().iter())
-                    .map(|(a, b)| a & b)
-                    .collect::<Vec<_>>();
-                Ok(<$type>::from((self.name(), result_bitmap)))
-            }
-
-            fn or(&self, rhs:Self) -> Self::Output {
-                let result_bitmap = self
-                    .values()
-                    .iter()
-                    .zip(rhs.values().iter())
-                    .map(|(a, b)| a | b)
-                    .collect::<Vec<_>>();
-                Ok(<$type>::from((self.name(), result_bitmap)))
-            }
-
-            fn xor(&self, rhs:Self) -> Self::Output {
-                let result_bitmap = self
-                    .values()
-                    .iter()
-                    .zip(rhs.values().iter())
-                    .map(|(a, b)| a ^ b)
-                    .collect::<Vec<_>>();
-                Ok(<$type>::from((self.name(), result_bitmap)))
+impl<T> DaftLogical<&DataArray<T>> for DataArray<T>
+where
+    T: DaftIntegerType,
+    T::Native: Ord + BitAnd<Output = T::Native> + BitOr<Output = T::Native> + BitXor<Output = T::Native>,
+{
+    type Output = DaftResult<Series>;
+
+    fn and(&self, rhs: &DataArray<T>) -> Self::Output {
+        bitwise_helper(self, rhs, |l, r| l & r)
+    }
+
+    fn or(&self, rhs: &DataArray<T>) -> Self::Output {
+        bitwise_helper(self, rhs, |l, r| l | r)
+    }
+
+    fn xor(&self, rhs: &DataArray<T>) -> Self::Output {
+        bitwise_helper(self, rhs, |l, r| l ^ r)
+    }
+}
+
+fn bitwise_helper<T, F>(
+    lhs: &DataArray<T>,
+    rhs: &DataArray<T>,
+    operation: F,
+) -> DaftResult<Series>
+where
+    T: DaftNumericType,
+    F: Fn(T::Native, T::Native) -> T::Native,
+{
+    match (lhs.len(), rhs.len()) {
+        (a, b) if a == b => {
+            let result_values: Vec<T::Native> = lhs
+                .as_arrow()
+                .values()
+                .iter()
+                .zip(rhs.as_arrow().values().iter())
+                .map(|(l, r)| operation(*l, *r))
+                .collect();
+
+            let result = DataArray::<T>::from((lhs.name(), result_values));
+            Ok(result.into_series())
```

Comment:
@colin-ho  : I'm getting a build error here  "error[E0599]: no method named `into_series` found for struct `array::DataArray` in the current scope
   --> src/daft-core/src/array/ops/comparison.rs:842:23
    |
842 |             Ok(result.into_series())
    |                       ^^^^^^^^^^^ method not found in `DataArray<T>`
    |
   ::: src/daft-core/src/array/mod.rs:25:1
    |
25  | pub struct DataArray<T: DaftPhysicalType> {
    | ----------------------------------------- method `into_series` not found for this struct
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `into_series`, perhaps you need to implement one of them:
            candidate #1: `IntoSeries`
            candidate #2: `SeriesLike`
"

Reply:
Attaching the build error for reference.  
[build_error_18_06_2024.txt](https://github.com/user-attachments/files/15876133/build_error_18_06_2024.txt)

Reply:
I think you might just need to bring the `IntoSeries` trait into scope.

Something like:

```
use crate::IntoSeries
```